quickjs.c

```diff
@@ -41945,9 +41947,7 @@ static uint64_t xorshift64star(uint64_t *pstate)
 
 static void js_random_init(JSContext *ctx)
 {
-    struct timeval tv;
-    gettimeofday(&tv, NULL);
-    ctx->random_state = ((int64_t)tv.tv_sec * 1000000) + tv.tv_usec;
+    getentropy(&ctx->random_state, sizeof(ctx->random_state));
```

Comment:
JS `Math.random` can be dropped. We want to ensure reproducible, predicted state. There is a near api to set random seed, which usually set as block height, is what contracts used